gcc/rust/hir/rust-ast-lower-item.h

```diff
@@ -53,6 +54,42 @@ class ASTLoweringItem : public ASTLoweringBase
     return resolver.translated;
   }
 
+  void visit (AST::ModuleBodied &module) override
+  {
+    auto crate_num = mappings->get_current_crate ();
+    Analysis::NodeMapping mapping (crate_num, module.get_node_id (),
+				   mappings->get_next_hir_id (crate_num),
+				   mappings->get_next_localdef_id (crate_num));
+
+    // should be lowered from module.get_vis()
+    HIR::Visibility vis = HIR::Visibility::create_public ();
```

Comment:
I think there is a case to make a much simpler HIR::visibility, an enum might just work: https://doc.rust-lang.org/stable/nightly-rustc/rustc_hir/hir/enum.VisibilityKind.html